hexes/hexes.py

```diff
@@ -51,6 +54,26 @@ def __repr__(self):
         ).format(s=self)
 
     @property
+    def text(self):
+        return self._text
+
+    @text.setter
+    def text(self, val):
+        dirty = val != self._text
```

Comment:
In fact the computation of `dirty` needs to be smarter than that anyways, so again, a descriptor seems warranted. Otherwise you get shit like

```python
a.text = "foo"
assert a.dirty
a.text = "foo"
assert not a.dirty
```